java/arcs/android/storage/database/AndroidSqliteDatabaseFactory.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.storage.database
+
+import android.content.Context
+import arcs.core.storage.database.Database
+import arcs.core.storage.database.DatabaseFactory
+import arcs.core.util.guardWith
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+/**
+ * [DatabaseFactory] implementation which constructs [DatabaseImpl] instances for use on Android
+ * with SQLite.
+ */
+class AndroidSqliteDatabaseFactory(context: Context) : DatabaseFactory {
```

Comment:
Optional, but I'd drop "Android" from the name. It's already in the android package.

Reply:
I had thought about that, but chose to stick with `Android`, because there could conceivably be more than one SqliteDatabaseFactory variant (e.g. on native platforms, or if JDBC was used or something), and I thought it would be less-than-optimal when developing a multiplexing DatabaseFactory to have to fully-qualify the class names.